python/cudf/dataframe/series.py

```diff
@@ -1075,7 +1075,10 @@ def hash_encode(self, stop, use_name=False):
         initial_hash = np.asarray(hash(self.name)) if use_name else None
         hashed_values = numerical.column_hash_values(
             self._column, initial_hash_values=initial_hash)
-        hashed_values = np.mod(hashed_values, stop)
+
+        # TODO: Binary op when https://github.com/rapidsai/cudf/pull/892 merged
+        cudautils.modulo(hashed_values.data.to_gpu_array(), stop)
```

Comment:
`hashed_values.data.to_gpu_array()` will return by copy, so you're writing into an ephemeral variable here

Reply:
If this were the case then shouldn't the test case fail: https://github.com/rapidsai/cudf/blob/1c0cd96b077c2703fe90fc99f5b8b916a672b04e/python/cudf/tests/test_dataframe.py#L1543

Reply:
If it isn't then we likely have a bug. Additionally, calling `to_gpu_array()` will skip null values by default so this won't work if you have a column of all nulls.